finite-state/shared/src/main/scala/fs2/data/esp/Tag.scala

```diff
@@ -49,10 +50,36 @@ object Tag {
         case Close => Iterator(Close)
         case Leaf  => Iterator(Leaf)
         case End   => Iterator(End)
-        case True  => Iterator(True)
         case _     => Iterator.empty
       }
 
   }
 
+  implicit def show[T: Show]: Show[Tag[T]] = {
+    case Input      => "$input"
+    case State(q)   => show"q$q"
+    case Depth(d)   => show"[$d]"
+    case Name(name) => show"<$name>"
+    case Open       => "<%>"
+    case Close      => "</%>"
+    case End        => "$"
+    case Leaf       => "%"
+    case Value(v)   => show"$v"
+  }
+
+  implicit def order[T: Order]: Order[Tag[T]] = Order.from {
```

Comment:
The non-enlightened among us could use a small comment on the desired order here.

Reply:
It is kind of arbitrary, I just needed an order for pretty printing matched cases in ESP so that I can compare them across different versions. I will just say it is an arbitrary one for internal use.

Reply:
I even made it private, as it is used only for pretty-printing for now.